linalg/hypre.cpp

```diff
@@ -804,7 +804,7 @@ HypreParMatrix::HypreParMatrix(MPI_Comm comm, HYPRE_BigInt glob_size,
    }
 #endif
 
-   hypre_MatvecCommPkgCreate(A);
+   if (!hypre_ParCSRMatrixCommPkg(A)) { hypre_MatvecCommPkgCreate(A); }
```

Comment:
I think the checks in the constructors are redundant.

Reply:
Done, thanks!